src/Features/markdown/Readme.tsx

```diff
@@ -59,36 +46,30 @@ const Readme = () => {
     setFile(combineParts());
   }, [parts]);
 
-  const onDragEnd: OnDragEndResponder = (res) => {
-    if (res.destination?.index === res.source.index) return;
-    const items = [...parts];
-    const [reorderedItem] = items.splice(res.source.index, 1);
-    items.splice(res.destination!.index, 0, reorderedItem);
-    setParts(items);
-  };
-
   return (
     <Stack
       style={{
         height: "100%",
       }}
     >
       <Group>
-        <Button onClick={openFile}>Open md file</Button>
-        <Button onClick={saveFile}>Save md file</Button>
         <Button
-          onClick={() => setShowPreview((p) => !p)}
-          disabled={!showEditor}
+          onClick={() =>
+            openFile((data) => {
+              setParts([{ title: "Content", template: data }]);
+            })
+          }
         >
-          {showPreview ? "Hide" : "Show"} preview
+          Open md file
         </Button>
-        <Button
-          onClick={() => setShowEditor((p) => !p)}
-          disabled={!showPreview}
-        >
-          {showEditor ? "Hide" : "Show"} editor
+        <Button onClick={saveFile}>Save md file</Button>
+        <Button onClick={() => toggle("preview")} disabled={!state.editor}>
+          {state.preview ? "Hide" : "Show"} preview
+        </Button>
+        <Button onClick={() => toggle("editor")} disabled={!state.preview}>
+          {state.editor ? "Hide" : "Show"} editor
         </Button>
-        {activeIndex ? (
+        {typeof activeIndex != "undefined" && parts?.length > 1 ? (
```

Comment:
`activeIndex ? x : y` covers activeIndex being undefined, null false 0 or "" as well, not sure why you changed it here

Reply:
Parts will also be an array forever (unless we are doing `setParts(null)` somewhere. 

```js
activeIndex && parts.length > 1 ? ()  : ()
```

Reply:
You're right about the initial check. Since `activeIndex` can be `0`, using `typeof activeIndex !== "undefined"` ensures the button shows for index `0` _(First array item within the section parts)_. 

As for `parts.length > 1`, the added question mark was indeed not needed there. Pushed a fix for this comment.